src/printer.ts

```diff
@@ -733,18 +758,56 @@ export class PugPrinter {
 		this.result += '\n';
 	}
 
-	private comment(token: CommentToken): string {
+	private comment(commentToken: CommentToken): string {
 		let result: string = this.computedIndent;
-		if (this.checkTokenType(this.previousToken, ['newline', 'indent', 'outdent'], true)) {
-			result += ' ';
-		}
-		result += '//';
-		if (!token.buffer) {
-			result += '-';
-		}
-		result += formatCommentPreserveSpaces(token.val, this.options.commentPreserveSpaces);
-		if (this.nextToken?.type === 'start-pipeless-text') {
-			this.pipelessComment = true;
+		if (/^\s*prettier-ignore\b/.test(commentToken.val)) {
```

Comment:
We should try to stick to the same behaviour as [typescript-eslint does with ban-ts-comment allow-with-description](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/ban-ts-comment.md#allow-with-description)

Reply:
ah good, wasn't aware of this. Will check!